src/components/TabQueryEditor.vue

```diff
@@ -197,11 +225,29 @@
           this.tab.unsavedChanges = false
         }
       },
+      escapeRegExp(string) {
+        return string.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&');
+      },
+      deparameterizeQuery() {
+        let query = this.editor.getValue()
+        _.each(this.queryParameterPlaceholders, param => {
+          query = query.replace(new RegExp(`(\\W|^)${this.escapeRegExp(param)}(\\W|$)`), `$1${this.queryParameterValues[param]}$2`)
+        });
+        return this.deparameterizeQueryResolve(query);
+      },
       async submitQuery() {
         this.running = true
         try {
+          let query = this.editor.getValue();
+
+          this.queryParameterPlaceholders = this.getQueryParameterPlaceholders()
```

Comment:
My feedback is mostly around doing things the 'vue way'.  Probably we need a hook into the codemirror API to inflate the `query` text into a data property when it changes.

If you feel like you want to take this on, then please feel free, otherwise I'll do it later this week. Broadly what I think needs to happen is:

1. Add a codemirror callback when the editor area changes and update a `query` property from `data`
2. Add computed properties for `queryParameterPlaceHolders`, and `deparameterizedQuery`
3. Move `escapeRegExp` to `data`

Reply:
Then you can make the `action` on `run` dynamic, so if queryParams > 0 'showmodal' else 'submit'